src/pint/models/dispersion_model.py

```diff
@@ -431,10 +431,12 @@ def remove_DMX_range(self, index):
             or isinstance(index, np.int64)
         ):
             indices = [index]
-        elif not isinstance(index, list) or not isinstance(index, np.ndarray):
+        elif not isinstance(index, (list, np.ndarray)):
```

Comment:
I'm a little confused by the wording but if I understand then you're just saying we should flip the logic around so that the `elif` should be a positive check if it is a list or array (and sets `indices = index`) and the `else` is the error?

Reply:
Just so. Sorry my answer put you through de Morgan's laws again...